wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitService.java

```diff
@@ -169,224 +191,281 @@ public RepoInfo clone(final CloneRequest request) throws URISyntaxException, Api
+    @GET
+    @Path("remote")
     @Produces({MediaType.APPLICATION_JSON, MediaType.TEXT_PLAIN})
-    public GenericEntity<List<Remote>> remoteList(RemoteListRequest request) throws ApiException {
+    public GenericEntity<List<Remote>> remoteList(@QueryParam("remoteName") String remoteName, @QueryParam("verbose") boolean verbose)
+            throws ApiException {
         try (GitConnection gitConnection = getGitConnection()) {
-            return new GenericEntity<List<Remote>>(gitConnection.remoteList(request)) {
+            return new GenericEntity<List<Remote>>(gitConnection.remoteList(remoteName, verbose)) {
             };
         }
     }
 
-    @Path("remote-update")
     @POST
+    @Path("remote")
     @Consumes(MediaType.APPLICATION_JSON)
     public void remoteUpdate(RemoteUpdateRequest request) throws ApiException {
         try (GitConnection gitConnection = getGitConnection()) {
-            gitConnection.remoteUpdate(request);
+            RemoteUpdateParams params = RemoteUpdateParams.create(request.getName())
+                                                          .withRemoveUrl(request.getRemoveUrl())
+                                                          .withRemovePushUrl(request.getRemovePushUrl())
+                                                          .withAddUrl(request.getAddUrl())
+                                                          .withAddPushUrl(request.getAddPushUrl())
+                                                          .withBranches(request.getBranches())
+                                                          .withAddBranches(request.isAddBranches());
+            gitConnection.remoteUpdate(params);
         }
     }
 
-    @Path("reset")
     @POST
+    @Path("reset")
     @Consumes(MediaType.APPLICATION_JSON)
     public void reset(ResetRequest request) throws ApiException {
         try (GitConnection gitConnection = getGitConnection()) {
-            gitConnection.reset(request);
-        }
-    }
-
-    @Path("rm")
-    @POST
-    @Consumes(MediaType.APPLICATION_JSON)
-    public void rm(RmRequest request) throws ApiException {
-        try (GitConnection gitConnection = getGitConnection()) {
-            gitConnection.rm(request);
+            ResetParams params = ResetParams.create(request.getCommit(), request.getType()).withFilePattern(request.getFilePattern());
+            gitConnection.reset(params);
         }
     }
 
+    @GET
     @Path("status")
-    @POST
     @Produces({MediaType.APPLICATION_JSON, MediaType.TEXT_PLAIN})
     public Status status(@QueryParam("format") StatusFormat format) throws ApiException {
         try (GitConnection gitConnection = getGitConnection()) {
             return gitConnection.status(format);
         }
     }
 
-    @Path("tag-create")
     @POST
+    @Path("tag")
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     public Tag tagCreate(TagCreateRequest request) throws ApiException {
```

Comment:
In other services we name methods in another way. WorkspaceService#createMachine, WorkspaceService#createSnapshot. Mb it would be better to rename this method to createTag. WDYT?